### ISSUE TITLE
Allow to opt-out from sandbox per app/action

### ISSUE THREAD

**Issue by quyse** [itchio/itch#756]
itch.io has not only games, but also apps. Certain apps like game development tools/editors may want to have access to full filesystem. For example, user may need to specify some file (texture, model) to be added into game map with editor app, and it would be easier if user could select the file in any directory, rather than firstly copy the file into `itch` user's home directory (or whatever place is available from sandbox).

As I understand right now sandbox could be enabled only for all or for nothing. I think it would be useful to allow explicitly enable or disable sandbox for certain apps, and/or even for particular actions of an app (e.g. sandbox disabled for `editor`, and enabled for `play`).

It also could be allowed to have `sandbox = false` setting in app manifest. I wouldn't like it to be a requirement, but it can simply ask user to disable sandbox (with all necessary warnings, and possibly even with explanation why it's needed provided in the manifest), so user can decide.

**Comment by fasterthanlime** [comment]
see also #751, "sandbox exceptions" should probably have their own issues.

Instead of having big disclaimers, I'd be in favor of whitelisting some game_ids, provided the developers sign their apps properly.

(It's _super easy_ to get users to click "yes" if that's the only thing that separates them from juicy juicy software, so I don't believe it'd be an efficient deterrent. Comparatively, faking app signatures is hard for attackers)

**Comment by quyse** [comment]
Well, as long as you allow user to disable sandbox completely for all apps, I think it's also pretty easy to convince users to do that. On the other hand, in case sandbox will become mandatory at some point, am I right that you're going to introduce some sort of approving system, similar to Google Play Store/AppStore, so some developers will be more equal than others? It doesn't sound cool, but I understand it could be necessary trade-off.

**Comment by fasterthanlime** [comment]
> On the other hand, in case sandbox will become mandatory at some point, am I right that you're going to introduce some sort of approving system, similar to Google Play Store/AppStore, so some developers will be more equal than others? It doesn't sound cool, but I understand it could be necessary trade-off.

One can release a full-scale, AAA game that runs perfectly well within the sandbox. If we are to set up a system where developers can be allowed sandbox exceptions (under some constraints, like signed binaries, etc.), then it'll be only for some software you mentioned, like text editors, image editors, etc. - I don't think that's a breach of itch.io's "openness" :)

**Comment by quyse** [comment]
> under some constraints, like signed binaries

If it means paying money to "certificate mafia", then it's a breach of itch.io's openness ;) Thinking a bit more about this, users also may be taught to run editors manually by finding and running appropriate executables by themselves. Or using developer version of itch app (current doc says "the sandbox might become mandatory (for app users)", therefore there will be developer version I guess). Any artificial limits will just unnecessary complicate things for people which wants to do their stuff. I think a reasonably scary warning before disabling sandbox is enough.

**Comment by quyse** [comment]
Alternatively an API for access to resources out of sandbox could be implemented, similar to browser's File API. Flatpak has just announced a similar API for user-controlled access to files/screenshots/printer/etc from inside a sandbox: https://blogs.gnome.org/mclasen/2016/07/08/portals-using-gtk-in-a-flatpak/ Although it might be too wide and over-complicated feature for itch.

**Comment by fasterthanlime** [comment]
> Although it might be too wide and over-complicated feature for itch.

Yeah, the problem here is that the "itch.io sandbox" is actually three distinct methods (one for each OS), and not all of them are as flexible as the others. For example, I'm not sure one can modify the sandbox policy of an app while it's running (macOS), nor the firejail profile (linux).

**Comment by fluffy-critter** [comment]
I just ran into this issue while trying to run Tiled on macOS, which mysteriously had no access to the files I was trying to work on. There's an easy workaround in simply launching the app manually instead of from the itch.io launcher, and frankly it's not that big a deal to just install tools the normal macOS way to be launched normally, but it was still a little puzzling why this was happening until I realized why.

**Comment by leafo** [comment]
You can now manage per-install settings from the manage dialog, including overriding the sandbox setting.
